Improve shader editor templates to be more descriptive

Calinou opened the pull request:
This makes the shader editor more self-explanatory out of the box.

Calinou committed:
Improve shader editor templates to be more descriptive

Calinou commented on editor/shader_create_dialog.cpp:
```diff
@@ -142,31 +142,66 @@ void ShaderCreateDialog::_create_new() {
-					code += "}\n";
+					code += R"(
+void sky() {
+	// Called for every texel in the sky texture.
```
Is this description correct?

clayjohn replied:
Not quite, it is kind of complex because sky shaders are used for cubemaps as well. See the doc for a detailed explanation
https://docs.godotengine.org/en/latest/tutorials/shaders/shader_reference/sky_shader.html

I'll try to think of a comment we can add that is short, but accurate

clayjohn replied:
```suggestion
	// Called for every visible pixel in the sky background as well as all pixels in the radiance cubemap.
```

Calinou commented on editor/shader_create_dialog.cpp:
```diff
@@ -142,31 +142,66 @@ void ShaderCreateDialog::_create_new() {
+					// Functions are ordered in the vertex-fragment-light order
+					// to follow established conventions.
+					code += R"(
+//void vertex() {
+	// Called for every vertex the material is visible on.
+//}
```
Should we uncomment `vertex()` by default? Does it have any performance cost if it's defined but left empty?

The same comment applies for CanvasItem shaders.

Chaosus replied:
I'm not sure is this a good idea - vertex shader is not used quite often as fragment.

Calinou commented:
I've modified the template for fog shaders (based on the [Fog shaders](https://docs.godotengine.org/en/latest/tutorials/shaders/shader_reference/fog_shader.html) manual page), and have also modified the sky shader comment accordingly.

clayjohn commented on editor/shader_create_dialog.cpp:
```diff
@@ -164,37 +164,74 @@ void ShaderCreateDialog::_create_new() {
+}
+
+//void light() {
+	// Called for every light affecting the material.
```
```suggestion
	// Called for every pixel for every light affecting the material.
```

The same goes for CanvasItem shaders.

clayjohn commented on editor/shader_create_dialog.cpp:
```diff
@@ -164,37 +164,76 @@ void ShaderCreateDialog::_create_new() {
-						code += "void fragment() {\n";
-						code += "\t// Place fragment code here.\n";
-						code += "}\n";
+						// `vertex()` and `light()` are commented out by default
+						// as these are used less often.
+						// Functions are ordered in the vertex-fragment-light order
+						// to follow established conventions.
```
I'm not sure the user will need these reminders every time they create a new shader. 

I would remove these comments. 

I would also have the functions un-commented by default. If they are empty, they don't do anything, so commenting them out doesn't change the generated shader at all.

clayjohn reviewed (approved):
Looks great!

YuriSizov commented:
Thanks!
